dtApi/src/app/speciality/modal-form/modal-form.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, Inject, OnInit } from '@angular/core'
+import {
+    MatDialog,
+    MatDialogRef,
+    MAT_DIALOG_DATA,
+} from '@angular/material/dialog'
+
+import { FormControl, FormGroup, Validators } from '@angular/forms'
+import { IsNumValidators } from '../isnum.validators'
+import { ApiService } from '../api.service'
+
+@Component({
+    selector: 'app-modal-form',
+    templateUrl: './modal-form.component.html',
+    styleUrls: ['./modal-form.component.scss'],
+})
+export class ModalFormComponent implements OnInit {
+    constructor(
+        private apiService: ApiService,
+        private dialogRef: MatDialogRef<ModalFormComponent>
+    ) {}
+
+    form = new FormGroup({
```

Comment:
use FormBuilder for reactive forms
https://angular.io/guide/reactive-forms#using-the-formbuilder-service-to-generate-controls